swim/memberlist.go

```diff
@@ -279,6 +281,83 @@ func (m *memberlist) SetLocalStatus(status string) {
 	m.postLocalUpdate()
 }
 
+func (m *memberlist) SetLocalLabel(key, value string) error {
+	// TODO implement a sane limit for the size of the labels to prevent users
+	// from impacting the performance of the gossip protocol.
+
+	// ensure that there is a labels map
+	if m.local.Labels == nil {
+		m.local.Labels = make(map[string]string)
+	}
+
+	// set the label
+	m.local.Labels[key] = value
+	m.postLocalUpdate()
```

Comment:
Maybe only when `key` is new or `value` is different to prevent unnecessary reincarnations?